[charts-pro] Add export menu to charts toolbar (#18210)
mui/mui-x: docs/data/charts/export/export.md

# Charts - Export [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

> <p class="description">Charts can be printed and exported as a PDF or an image.</p>

Reviewer: ```suggestion
<p class="description">Export charts as a PDF from the print dialog, or as an image.</p>
```

Export is available for the following charts: `<LineChartPro />`, `<BarChartPro />`, `<ScatterChartPro />`, `PieChartPro />`, `<Heatmap />`, `<FunnelChart />` and `<RadarChartPro />`.

## Enabling export

### Default toolbar

> Charts can be exported using the browser's native print dialog or as an image.

Reviewer: ```suggestion
```

> To enable this functionality, you need to enable the chart's toolbar by passing the `showToolbar` prop to the chart component.

Reviewer: ```suggestion
To enable exporting from the chart's toolbar, you need to enable it by passing the `showToolbar` prop to the chart component.
```

Export is available for the following charts: `<LineChartPro />`, `<BarChartPro />`, `<ScatterChartPro />`, `PieChartPro />`, `<Heatmap />`, `<FunnelChart />` and `<RadarChartPro />`.

## Overview

Charts can be exported using the browser's native print dialog or as an image.

To enable this functionality, you need to enable the chart's toolbar by passing the `showToolbar` prop to the chart component.

> The toolbar will then display an "Export" button that opens a menu with the export options.

Reviewer 2: I think you can follow completely the Data grid structure for export

- one section to say "Just add `showToolbar` prop to see the default options"
- one section to list the different export options

</codeblock>

## Export options

The following export options are available:

- [Print](/x/api/charts/chart-print-export-options/)

> - [Image export](/x/api/charts/chart-image-export-options/)

Reviewer: Maybe like this?

"Export behavior can be modified with [print](https://github.com/x/api/charts/chart-print-export-options/) and [image](https://github.com/x/api/charts/chart-image-export-options/) export options."

> In that case, the `imageExportOptions` must be an array of objects.

I hand't noticed it `must` be. Maybe we should allow a single string object as well? 🤔 Not sure...

In any way, I would probably rephrase to something in the lines of:

"In that case, the `imageExportOptions` must be an array of objects, each representing an image type to export."
